cmd/fscrypt/commands.go

```diff
@@ -107,7 +107,7 @@ var Encrypt = cli.Command{
 		immediately be used.`, directoryArg, shortDisplay(policyFlag),
 		shortDisplay(protectorFlag), mountpointArg),
 	Flags: []cli.Flag{policyFlag, unlockWithFlag, protectorFlag, sourceFlag,
-		userFlag, nameFlag, keyFileFlag, skipUnlockFlag, noRecoveryFlag},
+		userFlag, nameFlag, keyFileFlag, keyEncodedStrFlag, skipUnlockFlag, noRecoveryFlag},
```

Comment:
Command-line arguments shouldn't contain raw keys, as command lines are typically visible to other users on Linux.

The key will need to be passed on standard input instead (and then it can be binary rather than hex).  Take a look at what https://github.com/google/fscryptctl does.

Reply:
It makes sense.  Since in my case it won't be used by normal users but by IPC handled by root I added the CLI argument to make it easier.. I removed the argument from the pull request. So now it has to be entered on the standard input.